### PR TITLE
Custom package name in advanced options

### LINKED ISSUE #86: Custom package name in advanced options
Currently package name is always inferred from group and artifact ids.

### PR DESCRIPTION
**Pull request by tsegismont**
Closes #86

### CODE REVIEW

**Comment by tsegismont** [comment]
@jponge @danielpetisme can any of you please take a look?

**Comment by tsegismont** [comment]
That's a good point actually. At least we could report the issue nicely
instead of generating a invalid project (or failing with cryptic message)

Le lun. 25 mars 2019 à 18:51, Julien Ponge <notifications@github.com> a
écrit :

File: src/main/java/io/vertx/starter/service/GeneratorService.java [hunk 1]
```diff
@@ -174,6 +174,10 @@ private void createProject(VertxProject project, TempDir tempDir) throws IOExcep
   }
 
   private String packageName(VertxProject project) {
+    String packageName = project.getPackageName();
+    if (packageName != null && !packageName.trim().isEmpty()) {
```
Comment (danielpetisme) [inline src/main/java/io/vertx/starter/service/GeneratorService.java]: `StarterResource` defines a `isNotBlank` method. I'm sure is not the last time we'll have to do such snippet. Does it worth an `StringUtil` class?

**Comment by danielpetisme** [comment]
+1 for the naming guidelines/correction.
The classic red-bordered text box + message + a disabled button when the regex is invalid?

File: src/main/java/io/vertx/starter/service/GeneratorService.java (hunk 1 above)
Comment (tsegismont) [reply src/main/java/io/vertx/starter/service/GeneratorService.java]: Not unbearable yet to me :wink:

**Comment by tsegismont** [comment]
@danielpetisme not sure yet about the how. Will work on it today.

**Comment by tsegismont** [comment]
@jponge @danielpetisme can you take another look?

**Comment by danielpetisme** [comment]
Sweet!
Capture d’écran 2019-03-26 à 21 28 41

Some UI/UX concerns
- Use plus/minus icon to expand/collapse the advanced section (the chevron-right looks weird IMHO).
- Make the text "Advanced Options" clickable too (wrap everything into a border-less button?)
- Add a `<hr/>` to clearly mark the end of the advanced section (in order to anticipate the futur options)
- When an input is red a hint would be nice

**Comment by jponge** [comment]
I have no further comments :-)

**Comment by tsegismont** [comment]
I've updated the PR as per review. Will merge when Travis is ok. Please
open an issue if you have any more concern.

**Pull request merged**